lib/order_repository.rb

```diff
@@ -0,0 +1,118 @@
+require_relative './order'
+
+class OrderRepository
+
+    def all
+        # Executes the SQL query:
+        # SELECT id, customer_name, order_date, item_id FROM orders;
+        sql = 'SELECT id, customer_name, order_date, item_id FROM orders;'
+        params = []
+
+        result_set = DatabaseConnection.exec_params(sql, params)
+
+        orders = []
+        
+        result_set.each do |record|
+            order = Order.new
+            order.id = record['id'].to_i
+            order.customer_name = record['customer_name']
+            order.order_date = record['order_date']
+            order.item_id = record['item_id'].to_i
+
+            orders << order
+        end
+
+        return orders
+        # Returns an array of Order objects.
+    end
+
+    # Gets a single record by its ID
+    # One argument: the id (number)
+    def find(id)
+        # Executes the SQL query:
+        # SELECT id, customer_name, order_date, item_id FROM orders WHERE id = $1;
+        sql = 'SELECT id, customer_name, order_date, item_id FROM orders WHERE id = $1;'
+        params = [id]
+
+        result_set = DatabaseConnection.exec_params(sql, params)
+        record = result_set[0]
+            
+        order = Order.new
+        order.id = record['id'].to_i
+        order.customer_name = record['customer_name']
+        order.order_date = record['order_date']
+        order.item_id = record['item_id'].to_i
+
+        return order
+       
+    end
+
+    # Inserts new Order record into database
+    def create(order)
```

Comment:
This method is handling a lot of stuff. Could this be done in separate methods ?